sql/rpl_master.cc

```diff
@@ -1395,6 +1400,23 @@ bool show_binlogs(THD *thd) {
       }
     }
     protocol->store(file_length);
+
+    if (with_gtid) {
+      auto previous_gtid_set_map = mysql_bin_log.get_previous_gtid_set_map();
+      Sid_map sid_map(nullptr);
+      Gtid_set gtid_set(&sid_map, nullptr);
+      auto gtid_str = previous_gtid_set_map->at(std::string(fname));
```

Comment:
```suggestion
      const auto gtid_str = previous_gtid_set_map->at(std::string(fname));
```

Reply:
Explicit cast to `std::string` also seems redundant. `previous_gtid_set_map->at(fname);` should work.

Reply:
Also `std::map::at()` may throw an exception when `fname` is not in the map. Not handled here. I would use `map::find()` here instead
```c++
const auto gtid_str_it = previous_gtid_set_map->find(fname);
if (gtid_str_it != previous_gtid_set_map.end() && !gtid_str_it->second.empty()) {
...
```